### ISSUE TITLE
Identify number of companies which contain semicolon in either of the four scenario columns in `pstr_companies`

### ISSUE THREAD

**Issue by kalashsinghal** [2DegreesInvesting/tiltIndicator#405]
This issue relates to the 2nd point from this issue #401 

@Tilmon and @AnneSchoenauer 
Due to the presence of semicolon in either of the `weo_sector`, `weo_subsector`, `ipr_sector`, `ipr_subsector` columns in `pstr_companies` dataset, join with the same columns in `scenario` dataset didn't took place. Therefore, companies with such  a case needs to be either removed or PSTR code needs to be re-run again on these companies.  

UPDATE
 2. istr_company_level_results & pstr_company_level_results: Remove companies_id with ";" either of the scenario columns in pstr_companies input file. This case should already be captured by the removal of NA rows above, but as the problem origins from another source, I list it separately. Respective ticket identifying those cases can be found here: Identify number of companies which contain semicolon in either of the four scenario columns in pstr_companies

**Comment by kalashsinghal** [comment]
@Tilmon and @AnneSchoenauer Please have a look at this reprex which describe this case in detail: 

``` r
library(tidyverse)
library(readr)
library(here)
#> here() starts at /Users/kalashsinghal/git/2DII/tiltIndicator

pstr_comp <- read_csv(here("data-raw","ictr", "str_companies.csv"))

# Identify rows which contains semicolon in weo_sector, weo_subsector, 
# ipr_sector, ipr_subsector columns
weo_sec <- pstr_comp |>
  filter(str_detect(weo_sector, ";"))
weo_subsec <- pstr_comp |>
  filter(str_detect(weo_subsector, ";"))
ipr_sec <- pstr_comp |>
  filter(str_detect(ipr_sector, ";"))
ipr_subsec <- pstr_comp |>
  filter(str_detect(ipr_subsector, ";"))

bad <- rbind(weo_sec, weo_subsec, ipr_sec, ipr_subsec) |>
  distinct()

# Number of unique companies in with such bad cases 
length(unique(bad$companies_id))
#> [1] 48486

# Number of unique companies in `pstr_dataset`
length(unique(pstr_comp$companies_id))
#> [1] 287978

#To remove these companies from `pstr_companies` dataset 
final <- pstr_comp |> filter(!companies_id %in% unique(bad$companies_id))
final
#> # A tibble: 455,425 × 11
#>    companies_id        company_name clustered activity_uuid_produc…¹ isic_4digit
#>    <chr>               <chr>        <chr>     <chr>                  <chr>      
#>  1 %ef%bb%bfmathias-m… mathias mas… <NA>      <NA>                   <NA>       
#>  2 -fred-sl_000000054… + fred s.l   fish, de… 26104519-4d49-5d85-bc… 0311       
#>  3 -fred-sl_000000054… + fred s.l   fish, fr… 0fe31e67-346a-504c-a0… 0311       
#>  4 -julia-played-here… * julia pla… <NA>      <NA>                   <NA>       
#>  5 010shop-grosshande… 010-shop / … indoor p… 0f2e967b-dd3e-509f-9d… 0161       
#>  6 010shop-grosshande… 010-shop / … <NA>      <NA>                   <NA>       
#>  7 0110-architecten-b… 01-10 archi… <NA>      <NA>                   <NA>       
#>  8 024h-schlusseldien… @0-24h schl… <NA>      <NA>                   <NA>       
#>  9 02el0ftravel-gmbh-… 02el0f.trav… bus       1a0a4a2e-b47e-5c4c-a8… 4921       
#> 10 02el0ftravel-gmbh-… 02el0f.trav… <NA>      <NA>                   <NA>       
#> # ℹ 455,415 more rows
#> # ℹ abbreviated name: ¹​activity_uuid_product_uuid
#> # ℹ 6 more variables: tilt_sector <chr>, tilt_subsector <chr>,
#> #   ipr_sector <chr>, ipr_subsector <chr>, weo_sector <chr>,
#> #   weo_subsector <chr>
```

<sup>Created on 2023-06-05 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

In total there are 48486 companies out of total 287978 which falls under this case.

**Comment by maurolepore** [comment]
@Tilmon and @kalashsinghal, 

I'm confused. For each company with this issue, what do you want to do?:

1. Remove all rows (all of these companies will be gone)
2. Remove the rows where the relevant columns have ";" and leave any other remaining row if there is any (some of these companies may be gone).
3. Same as b but if if all rows have this issue leave at least one row fulll of `NA`.

Also, if I understand correctly you want to use the input (`pstr_companies`) to identify the problematic rows but then remove them from the company-level output. However above I see @kalashsinghal shows how to remove these companies from the input `pstr_companies`. Is this intentional?

And finally, the overview issue #401 says that _This case should already be captured by the removal of NA rows above_ but the item "above" removes `NA` from the product-level output so the company-level output should still need this fix, right?

**Comment by Tilmon** [comment]
hi @maurolepore and @kalashsinghal , sorry for the confusion. Now that I re-read through my description, I understand it's indeed confusing. Let me clarify: 

On both product- and company-level, we want to do the following: 
- We want to remove `NA `rows from the final output
- If a company has only `NAs`, then we want to keep that company with ONLY 1 row that shows NA for all columns. For this case, @kalashsinghal provided a solution [here](https://github.com/2DegreesInvesting/tiltIndicator/issues/402#issuecomment-1576708618).  On company-level, @maurolepore already implemented his own "version" of that solution in [Fix XSTR outputs (#396)](https://github.com/2DegreesInvesting/tiltIndicator/blob/396_Fix-xstr-outputs/data-raw/396_fix-xstr-outputs.md). However, in that script, it's still missing on product-level. That I did yesterday on my PC with the code below that @kalashsinghal provided. 
- Now 1 one of the origins of `NAs `seems to be the ";" in the scenario-sector columns. That's why, originally, we were working on a solution to exclude these companies explicitly. After now having detected some other issues and decided to remove `NA `rows in general, I think these specific ";"-cases are already removed from the output anyway if we remove `NAs`. Is that correct? If so, I suggest to consider this case as closed and I will later create a ticket for "enhancements" to solve the ";" issue at the source of the `pstr_companies `dataset. 

Do you agree? If so, pls close this issue. 
 
```{r}
no_NAs <- pstr_product_level |>
  filter(!is.na(risk_category)) 

# select company ids where all `risk_category` is only `NA`
ids <- pstr_product_level |>
  group_by(companies_id) |>
  filter(all(is.na(risk_category))) |>
  distinct(companies_id)

pstr_product_level <- bind_rows(no_NAs, ids)
```

**Comment by maurolepore** [comment]
@Tilmon, I'll close this issue as done as a patch to the output we already delivered. 

The issue to fix the problem from the source is https://github.com/2DegreesInvesting/tiltIndicator/issues/439.
